File: CS/ReadOnlyObject.cs
using System;
using DevExpress.Xpo;

namespace ReadOnlyApp {
	[NonPersistent]
	public class ReadOnlyObject : XPLiteObject {
        public ReadOnlyObject(Session session) : base(session) { }

		// prevents object saving via BindingManagerBase.EndCurrentEdit
        protected override void DoEndEditAction() { }
	}
}

File: VB/ReadOnlyObject.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace ReadOnlyApp
	<NonPersistent> _
	Public Class ReadOnlyObject
		Inherits XPLiteObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		' prevents object saving via BindingManagerBase.EndCurrentEdit
		Protected Overrides Sub DoEndEditAction()
		End Sub
	End Class
End Namespace
